sqe-http-api/Controllers/UserController.cs

```diff
@@ -20,34 +23,126 @@ public UserController(IUserService userServiceAuthenticate)
+            }
+        }
+        
+        /// <summary>
+        /// Confirms creation of new user account.
+        /// </summary>
+        /// <param name="payload">JSON object with token from user registration email.</param>
+        /// <returns>Returns 200 for success and 404 for failure.</returns>
+        [AllowAnonymous]
+        [HttpPost("confirm-registration")]
+        public async Task<ActionResult<UserDTO>> ConfirmUserRegistration([FromBody] EmailTokenDTO payload)
+        {
+            try
+            {
+                await _userService.ConfirmUserRegistrationAsync(payload.token);
+                return Ok();
+            }
+            catch
+            {
+                return NotFound();
+            }
+        }
 
         /// <summary>
         /// Provides a JWT bearer token for valid username and password
         /// </summary>
         /// <param Name="userParam">JSON object with a username and password parameter</param>
         [AllowAnonymous]
-        [HttpPost("login")] // api/v1/user/login
+        [HttpPost("login")]
         public async Task<ActionResult<LoginResponseDTO>> AuthenticateAsync([FromBody]LoginRequestDTO userParam)
         {
             var user = await _userService.AuthenticateAsync(userParam.userName, userParam.password);
             if (user == null)
-                return Unauthorized(new { message = "Username or password is incorrect", code = 600}); // TODO: Add Error Code
+                return Unauthorized(new { message = "Username or password is incorrect ", code = 600}); // TODO: Add Error Code
 
             return user;
         }
-
+        
         /// <summary>
-        /// Provides the user details for a user with valid JWT in the Authorize header
+        /// Sends secret token to user's email to allow password reset.
         /// </summary>
-        [HttpGet] // api/v1/user
-        public ActionResult<UserDTO> GetCurrentUser()
+        /// <param name="payload">JSON object with the email address for the user who wants to reset a lost password.</param>
+        /// <returns></returns>
+        [AllowAnonymous]
+        [HttpPost("forgot-password")]
+        public async Task<ActionResult> ForgotPassword([FromBody] ResetUserPasswordRequestDTO payload)
         {
-            var user = _userService.GetCurrentUser();
-
-            if (user == null)
-                return Unauthorized(new { message = "No current user", code =601 }); // TODO: Add Error Code
-
-            return user;
+            await _userService.RequestResetLostPasswordAsync(payload.email);
+            return NoContent();
+        }
+        
+        /// <summary>
+        /// Change the password for the currently logged in user.
+        /// </summary>
+        /// <param name="payload">A JSON object with the old password and the new password.</param>
+        /// <returns>Status 200 with a successful request or status 409 for an unsuccessful request</returns>
```

Comment:
Again, return 204 for empty successful queries

Reply:
Thanks, I have updated the code and the description in our Website API Google Doc.